### ISSUE TITLE
Netty automatically closes the stream.

### ISSUE THREAD

**Issue by mdsuhaibsa** [reactor/reactor-netty#2852]
Ealier we raised a ticket #2735, 

We manually put those fixes inside our application. Using it, we tested the application with high traffic(210K MPS). We can see netty automatically closes the stream for around 40% of incoming messages. Without those fixes in the application we are seeing no impact on the incoming messages.

Traceback which we can see in the application:

java.util.concurrent.CancellationException\n\tat java.base/java.util.concurrent.CompletableFuture.cancel(CompletableFuture.java:2478)\n\tat reactor.core.publisher.MonoCompletionStage$1.cancel(MonoCompletionStage.java:60)\n\tat reactor.core.publisher.Operators.terminate(Operators.java:1277)\n\tat reactor.core.publisher.MonoFlatMap$FlatMapMain.cancel(MonoFlatMap.java:187)\n\tat reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.cancel(FluxOnAssembly.java:654)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drainLoop(Operators.java:2254)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drain(Operators.java:2222)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.cancel(Operators.java:2034)\n\tat reactor.core.publisher.Operators.terminate(Operators.java:1277)\n\tat reactor.core.publisher.MonoFlatMap$FlatMapInner.cancel(MonoFlatMap.java:272)\n\tat reactor.core.publisher.MonoFlatMap$FlatMapMain.cancel(MonoFlatMap.java:188)\n\tat reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.cancel(FluxOnAssembly.java:654)\n\tat reactor.core.publisher.FluxDoOnEach$DoOnEachSubscriber.cancel(FluxDoOnEach.java:113)\n\tat reactor.core.publisher.FluxPeek$PeekSubscriber.cancel(FluxPeek.java:153)\n\tat reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.cancel(FluxOnAssembly.java:654)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drainLoop(Operators.java:2254)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drain(Operators.java:2222)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.cancel(Operators.java:2034)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drainLoop(Operators.java:2254)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drain(Operators.java:2222)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.cancel(Operators.java:2034)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drainLoop(Operators.java:2254)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drain(Operators.java:2222)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.cancel(Operators.java:2034)\n\tat reactor.core.publisher.MonoPeekTerminal$MonoTerminalPeekSubscriber.cancel(MonoPeekTerminal.java:144)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drainLoop(Operators.java:2254)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drain(Operators.java:2222)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.cancel(Operators.java:2034)\n\tat reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.cancel(MonoIgnoreThen.java:143)\n\tat reactor.core.publisher.MonoPeekTerminal$MonoTerminalPeekSubscriber.cancel(MonoPeekTerminal.java:144)\n\tat reactor.core.publisher.MonoPeekTerminal$MonoTerminalPeekSubscriber.cancel(MonoPeekTerminal.java:144)\n\tat reactor.core.publisher.Operators.terminate(Operators.java:1277)\n\tat reactor.netty.channel.ChannelOperations.terminate(ChannelOperations.java:474)\n\tat reactor.netty.http.server.HttpServerOperations.onInboundClose(HttpServerOperations.java:677)\n\tat reactor.netty.channel.ChannelOperationsHandler.channelInactive(ChannelOperationsHandler.java:73)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:305)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:281)\n\tat io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:274)\n\tat io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1405)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:301)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:281)\n\tat io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:901)\n\tat io.netty.handler.codec.http2.AbstractHttp2StreamChannel$Http2ChannelUnsafe$2.run(AbstractHttp2StreamChannel.java:737)\n\tat io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)\n\tat io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)\n\tat io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)\n\tat io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:403)\n\tat io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)\n\tat io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)\n\tat io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n

reactor-netty version:
+--- org.springframework.boot:spring-boot-starter-webflux:2.7.10
|    +--- org.springframework.boot:spring-boot-starter:2.7.10 (*)
|    +--- org.springframework.boot:spring-boot-starter-json:2.7.10 (*)
|    +--- org.springframework.boot:spring-boot-starter-reactor-netty:2.7.10
|    |    \--- io.projectreactor.netty:reactor-netty-http:1.0.30

**Comment by violetagg** [comment]
@mdsuhaibsa Please provide a reproducible example.

**Comment by mdsuhaibsa** [comment]
The above issue happens only on higher traffic of around 210k messages per second.
However, on top of library,
**io.projectreactor.netty:reactor-netty-http:1.0.30**, we applied two custom files from the bug fix #2735 ,

1. HttpServerConfig.java from https://github.com/reactor/reactor-netty/pull/2758/files
2. ServerTransport.java from https://github.com/reactor/reactor-netty/pull/2777/files

Using the custom applied fix we are seeing issues. But when we remove the custom files, it works. 
Can you please look into this direction since the problem is happening on higher traffic?

**Comment by violetagg** [comment]
@mdsuhaibsa You have to provide at least some complete log file, showing the requests that are closed wrong. Just with one stack trace, it will be really hard to do a proper investigation!

**Comment by mdsuhaibsa** [comment]
It is nearly impossible to enable debug log at such a high traffic. Please elaborate on what type of log you are looking for. Is there a custom way to get that details ?

**Comment by mdsuhaibsa** [comment]
@violetagg 
Under which circumstance the following API can get invoked ?
-- fireChannelInactive()
In our case,
our application just accept the request from consumer and waits for the producer to respond (before responding to consumer)
within few milliseconds the connection get terminated from Netty end with the above traceback
Any pointer on what can cause this trigger ?

**Comment by violetagg** [comment]
> @violetagg
> Under which circumstance the following API can get invoked ?
> -- fireChannelInactive()
> In our case,
> our application just accept the request from consumer and waits for the producer to respond (before responding to consumer)
> within few milliseconds the connection get terminated from Netty end with the above traceback
> Any pointer on what can cause this trigger ?

This is connection close, this can be initiated by the client or the server, in both cases it will result in channel inactive.

**Comment by mdsuhaibsa** [comment]
@violetagg 
Here are the debug logs:

**Success Case:**

{"instant":{"epochSecond":1689067180,"nanoOfSecond":167030749},"thread":"epollEventLoopGroup-2-6","level":"DEBUG","loggerName":"reactor.netty.http.server.h2","message":"[id: 0x9ce02fe6, L:/10.233.111.209:8080 - R:/127.0.0.6:52715] INBOUND HEADERS: streamId=1463181 headers=DefaultHttp2Headers[:authority: application:8000, :method: POST, :path: /application/path, :scheme: http, accept: application/json, content-type: application/json] padding=0 endStream=false","endOfBatch":false,"loggerFqcn":"io.netty.util.internal.logging.LocationAwareSlf4JLogger","threadId":535,"threadPriority":10,"messageId":"","messageTimestamp":"2023-07-11T09:19:40.167+0000","application":"temp","microservice":"application","namespace":"test-ns","node":"k8s-node-11.moon.lab.in.test.com","subsystem":"router","instanceType":"prod","processId":"1"}
{"instant":{"epochSecond":1689067180,"nanoOfSecond":167473550},"thread":"epollEventLoopGroup-2-6","level":"DEBUG","loggerName":"reactor.netty.http.server.h2","message":"[id: 0x9ce02fe6, L:/10.233.111.209:8080 - R:/127.0.0.6:52715] INBOUND DATA: streamId=1463181 padding=0 endStream=false length=4025 bytes=7b0a202020202270647553657373696f6e4964223a20223233343737222c0a2020202022736c696365496e666f223a202261646667313237222c0a2020202022...","endOfBatch":false,"loggerFqcn":"io.netty.util.internal.logging.LocationAwareSlf4JLogger","threadId":535,"threadPriority":10,"messageId":"","messageTimestamp":"2023-07-11T09:19:40.167+0000","application":"temp","microservice":"application","namespace":"test-ns","node":"k8s-node-11.moon.lab.in.test.com","subsystem":"router","instanceType":"prod","processId":"1"}
{"instant":{"epochSecond":1689067180,"nanoOfSecond":167562678},"thread":"epollEventLoopGroup-2-6","level":"DEBUG","loggerName":"reactor.netty.http.server.h2","message":"[id: 0x9ce02fe6, L:/10.233.111.209:8080 - R:/127.0.0.6:52715] INBOUND DATA: streamId=1463181 padding=0 endStream=true length=0 bytes=","endOfBatch":false,"loggerFqcn":"io.netty.util.internal.logging.LocationAwareSlf4JLogger","threadId":535,"threadPriority":10,"messageId":"","messageTimestamp":"2023-07-11T09:19:40.167+0000","application":"temp","microservice":"application","namespace":"test-ns","node":"k8s-node-11.moon.lab.in.test.com","subsystem":"router","instanceType":"prod","processId":"1"}
{"instant":{"epochSecond":1689067180,"nanoOfSecond":192000149},"thread":"epollEventLoopGroup-2-6","level":"DEBUG","loggerName":"reactor.netty.http.server.h2","message":"[id: 0x9ce02fe6, L:/10.233.111.209:8080 - R:/127.0.0.6:52715] OUTBOUND HEADERS: streamId=1463181 headers=DefaultHttp2Headers[:status: 200, content-type: text/plain; charset=utf-8, date: Tue, 11 Jul 2023 09:19:40 GMT, content-length: 18] padding=0 endStream=false","endOfBatch":false,"loggerFqcn":"io.netty.util.internal.logging.LocationAwareSlf4JLogger","threadId":535,"threadPriority":10,"messageId":"","messageTimestamp":"2023-07-11T09:19:40.192+0000","application":"temp","microservice":"application","namespace":"test-ns","node":"k8s-node-11.moon.lab.in.test.com","subsystem":"router","instanceType":"prod","processId":"1"}
{"instant":{"epochSecond":1689067180,"nanoOfSecond":192197912},"thread":"epollEventLoopGroup-2-6","level":"DEBUG","loggerName":"reactor.netty.http.server.h2","message":"[id: 0x9ce02fe6, L:/10.233.111.209:8080 - R:/127.0.0.6:52715] OUTBOUND DATA: streamId=1463181 padding=0 endStream=true length=18 bytes=7b44656661756c7420526573706f6e73657d","endOfBatch":false,"loggerFqcn":"io.netty.util.internal.logging.LocationAwareSlf4JLogger","threadId":535,"threadPriority":10,"messageId":"","messageTimestamp":"2023-07-11T09:19:40.192+0000","application":"temp","microservice":"application","namespace":"test-ns","node":"k8s-node-11.moon.lab.in.test.com","subsystem":"router","instanceType":"prod","processId":"1"}

**Failure Case**

{"instant":{"epochSecond":1689067175,"nanoOfSecond":222951597},"thread":"epollEventLoopGroup-2-6","level":"DEBUG","loggerName":"reactor.netty.http.server.h2","message":"[id: 0x9ce02fe6, L:/10.233.111.209:8080 - R:/127.0.0.6:52715] INBOUND HEADERS: streamId=1460953 headers=DefaultHttp2Headers[:authority: application:8000, :method: POST, :path: /application/path, :scheme: http, content-type: application/json] padding=0 endStream=false","endOfBatch":false,"loggerFqcn":"io.netty.util.internal.logging.LocationAwareSlf4JLogger","threadId":535,"threadPriority":10,"messageId":"","messageTimestamp":"2023-07-11T09:19:35.222+0000","namespace":"test-ns","node":"k8s-node-11.moon.lab.in.test.com",,"subsystem":"router","instanceType":"prod","processId":"1"}
{"instant":{"epochSecond":1689067175,"nanoOfSecond":223523429},"thread":"epollEventLoopGroup-2-6","level":"DEBUG","loggerName":"reactor.netty.http.server.h2","message":"[id: 0x9ce02fe6, L:/10.233.111.209:8080 - R:/127.0.0.6:52715] INBOUND DATA: streamId=1460953 padding=0 endStream=false length=4025 bytes=7b0a202020202270647553657373696f6e4964223a20223233343737222c0a2020202022736c696365496e666f223a202261646667313237222c0a2020202022...","endOfBatch":false,"loggerFqcn":"io.netty.util.internal.logging.LocationAwareSlf4JLogger","threadId":535,"threadPriority":10,"messageId":"","messageTimestamp":"2023-07-11T09:19:35.223+0000","namespace":"test-ns","node":"k8s-node-11.moon.lab.in.test.com",,"subsystem":"router","instanceType":"prod","processId":"1"}
{"instant":{"epochSecond":1689067175,"nanoOfSecond":223598033},"thread":"epollEventLoopGroup-2-6","level":"DEBUG","loggerName":"reactor.netty.http.server.h2","message":"[id: 0x9ce02fe6, L:/10.233.111.209:8080 - R:/127.0.0.6:52715] INBOUND DATA: streamId=1460953 padding=0 endStream=true length=0 bytes=","endOfBatch":false,"loggerFqcn":"io.netty.util.internal.logging.LocationAwareSlf4JLogger","threadId":535,"threadPriority":10,"messageId":"","messageTimestamp":"2023-07-11T09:19:35.223+0000","namespace":"test-ns","node":"k8s-node-11.moon.lab.in.test.com",,"subsystem":"router","instanceType":"prod","processId":"1"}
{"instant":{"epochSecond":1689067176,"nanoOfSecond":218988615},"thread":"epollEventLoopGroup-2-6","level":"DEBUG","loggerName":"reactor.netty.http.server.h2","message":"[id: 0x9ce02fe6, L:/10.233.111.209:8080 - R:/127.0.0.6:52715] INBOUND RST_STREAM: streamId=1460953 errorCode=0","endOfBatch":false,"loggerFqcn":"io.netty.util.internal.logging.LocationAwareSlf4JLogger","threadId":535,"threadPriority":10,"messageId":"","messageTimestamp":"2023-07-11T09:19:36.218+0000","namespace":"test-ns","node":"k8s-node-11.moon.lab.in.test.com",,"subsystem":"router","instanceType":"prod","processId":"1"}

Can you please provide on some input on what's happening?

**Comment by mdsuhaibsa** [comment]
@violetagg Can you provide us some input please?

**Comment by violetagg** [comment]
@mdsuhaibsa With the provided logs/stacktrace I cannot guess what might be the problem. When the mentioned fixe were made they were tested with all other Reactor Netty fixes in place also with a concrete Netty version. As you use pretty new version (1.0.30) I suggest that you update to (1.0.34) so that we are sure that all Reactor Netty/Netty fixes are in place.

**Comment by mdsuhaibsa** [comment]
@violetagg  Could you please tell us what logs are needed. I will try to collect.

**Comment by violetagg** [comment]
> @violetagg Could you please tell us what logs are needed. I will try to collect.

Can you collect logs for this connection (not only this stream)? We need to know the `lastStreamId` sent with go_away, whether there are some exception etc.

**Comment by mdsuhaibsa** [comment]
@violetagg 
After changing the io.projectreactor:reactor-core version from 3.4.28 to 3.4.18, the issue is no longer seen.
io.projectreactor:reactor-core:3.4.28 -> 3.4.18 (Dependency attached)
We also tested with 3.4.31 still the issue persists. 

We are using following netty version:
+--- org.springframework.boot:spring-boot-starter-webflux:2.7.10
| +--- org.springframework.boot:spring-boot-starter:2.7.10 ()
| +--- org.springframework.boot:spring-boot-starter-json:2.7.10 ()
| +--- org.springframework.boot:spring-boot-starter-reactor-netty:2.7.10
| | --- io.projectreactor.netty:reactor-netty-http:1.0.30

[dependency_tree.txt](https://github.com/reactor/reactor-netty/files/12147387/dependency_tree.txt)

Could you please tell is what could be the difference b/w 3.4.18 and 3.4.28 which is triggering channel incative under high [traffic?](url)
**Stacktrace pasted below**
`java.util.concurrent.CancellationException\n\tat java.base/java.util.concurrent.CompletableFuture.cancel(CompletableFuture.java:2478)\n\tat reactor.core.publisher.MonoCompletionStage$1.cancel(MonoCompletionStage.java:60)\n\tat reactor.core.publisher.Operators.terminate(Operators.java:1277)\n\tat reactor.core.publisher.MonoFlatMap$FlatMapMain.cancel(MonoFlatMap.java:187)\n\tat reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.cancel(FluxOnAssembly.java:654)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drainLoop(Operators.java:2254)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drain(Operators.java:2222)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.cancel(Operators.java:2034)\n\tat reactor.core.publisher.Operators.terminate(Operators.java:1277)\n\tat reactor.core.publisher.MonoFlatMap$FlatMapInner.cancel(MonoFlatMap.java:272)\n\tat reactor.core.publisher.MonoFlatMap$FlatMapMain.cancel(MonoFlatMap.java:188)\n\tat reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.cancel(FluxOnAssembly.java:654)\n\tat reactor.core.publisher.FluxDoOnEach$DoOnEachSubscriber.cancel(FluxDoOnEach.java:113)\n\tat reactor.core.publisher.FluxPeek$PeekSubscriber.cancel(FluxPeek.java:153)\n\tat reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.cancel(FluxOnAssembly.java:654)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drainLoop(Operators.java:2254)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drain(Operators.java:2222)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.cancel(Operators.java:2034)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drainLoop(Operators.java:2254)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drain(Operators.java:2222)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.cancel(Operators.java:2034)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drainLoop(Operators.java:2254)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drain(Operators.java:2222)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.cancel(Operators.java:2034)\n\tat reactor.core.publisher.MonoPeekTerminal$MonoTerminalPeekSubscriber.cancel(MonoPeekTerminal.java:144)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drainLoop(Operators.java:2254)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.drain(Operators.java:2222)\n\tat reactor.core.publisher.Operators$MultiSubscriptionSubscriber.cancel(Operators.java:2034)\n\tat reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.cancel(MonoIgnoreThen.java:143)\n\tat reactor.core.publisher.MonoPeekTerminal$MonoTerminalPeekSubscriber.cancel(MonoPeekTerminal.java:144)\n\tat reactor.core.publisher.MonoPeekTerminal$MonoTerminalPeekSubscriber.cancel(MonoPeekTerminal.java:144)\n\tat reactor.core.publisher.Operators.terminate(Operators.java:1277)\n\tat reactor.netty.channel.ChannelOperations.terminate(ChannelOperations.java:474)\n\tat reactor.netty.http.server.HttpServerOperations.onInboundClose(HttpServerOperations.java:677)\n\tat reactor.netty.channel.ChannelOperationsHandler.channelInactive(ChannelOperationsHandler.java:73)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:305)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:281)\n\tat io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:274)\n\tat io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1405)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:301)\n\tat io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:281)\n\tat io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:901)\n\tat io.netty.handler.codec.http2.AbstractHttp2StreamChannel$Http2ChannelUnsafe$2.run(AbstractHttp2StreamChannel.java:737)\n\tat io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)\n\tat io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)\n\tat io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)\n\tat io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:403)\n\tat io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)\n\tat io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)\n\tat io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n`

**Comment by pderop** [comment]
@mdsuhaibsa ,

I cannot comment about the dependency change you did regarding reactor-core, because I don't see the relation between your current problem and reactor-core, and I tend to think it is unrelated.

I have some comments:

1) why cannot you upgrade to 1.0.34, where all fixes have been regrouped and properly validated ?

2) if you really can't upgrade to latest 1.0.34, then please attach to this issue a tar of the files that you have patched in reactor netty 1.0.30 version

3) in addition, please provide the informations that Violeta previously requested, see https://github.com/reactor/reactor-netty/issues/2852#issuecomment-1645023878; indeed, there might be some interesting exceptions, like for example :

```
19:35:54.834 [reactor-http-nio-6] WARN  io.netty.util.ReferenceCountUtil - Failed to release a message: PooledUnsafeDirectByteBuf(freed)
io.netty.util.IllegalReferenceCountException: refCnt: 0, decrement: 1
```

thanks.

**Comment by pderop** [comment]
I'm closing this issue for the moment, but If you can provide the needed informations, then we can reopen it.

We need:
- the logs asked by Violeta in https://github.com/reactor/reactor-netty/issues/2852#issuecomment-1645023878
- the tar.gz of the files that you have patched in reactor netty 1.0.30
- Also, I encourage to use Netty 4.1.93.Final version, which is providing the PR #13388 that is fixing an issue related to HTTP/2 stream close under high traffic, it may resolve the issue.

thank you.
